### PR TITLE
xc7: add clock tiles BUFG and BUFH

### PR DESCRIPTION
**Pull request by acomodi**
This PR is part of https://github.com/SymbiFlow/symbiflow-arch-defs/pull/1060 and adds the BUFH and BUFG tile types.

Once this lands the next patches can be added to have a working ROI-less and clock-tree driven implementations.

### CODE REVIEW

File: xc7/make/arch_define.cmake [hunk 1]
```diff
@@ -83,6 +83,7 @@ function(ADD_XC7_ARCH_DEFINE)
         --pcf \${INPUT_IO_FILE} \
         --eblif \${OUT_EBLIF} \
         --top \${TOP} \
+        --allow_orphan_sinks \
```
Comment (litghost) [inline xc7/make/arch_define.cmake]: Can we try dropping this option?  I don't believe it is required anymore for fasm2bels to work.

Comment (acomodi) [reply xc7/make/arch_define.cmake]: Done, should I locally check that fasm2bels works locally or is it already tested in CI?

Comment (litghost) [reply xc7/make/arch_define.cmake]: Locally.  We cannot test ROI-less stuff until the lookahead search is updated.

**Comment by litghost** [comment]
Let's reword the commit message on 3022931c524885338f7556b56a4df83a3424bd83, and maybe break up the commits.

**Pull request merged**